File: DataStructures/LinkedList/LinkedList.java
public class LinkedList<T> {

    private Node root;

    public LinkedList(T value ){
        root = new Node(value);
    }

    public LinkedList(){
        root = null;
    }


    public LinkedList(T[] arrayO){
     if(arrayO.length == 0){
         root = null;
     }
     else{
         root = new Node(arrayO[0]);
         for(int i = 1; i < arrayO.length; i++){
             AddValue(arrayO[i]);
         }
     }
    }

    public boolean removeValue(T value){
        if (root == null) return false;

        if(root.value == value){
            root = root.descendant;
            return true;
        }
        else{
            Node previousNode = root;
            Node node = root.descendant;
            while(node != null){
                if (node.value == value){
                    previousNode.descendant = node.descendant;
                    return true;
                }
                previousNode = node;
                node = node.descendant;
            }
            return false;
        }


    }
    public void addToBeginning(T value){
        Node node = new Node(value);
        node.descendant = root;
        root = node;
    }

    public void removeAllOccurencies(T value){
        while(removeValue(value)){

        }
    }


    public void printValues(){
        Node node = root;
        while(node != null){
            System.out.print(node.value + " ");
            node = node.descendant;
        }
        System.out.println();
    }

    public void AddValue(T value){
        if (root == null){
            root = new Node(value);
            return;
        }
        Node n = root;
        while(n.descendant != null){
            n = n.descendant;
        }
        n.descendant = new Node(value);
    }
    public boolean Search(T value){
        Node node = root;
        while(node != null){
            if(node.value.equals(value)){
                return true;
            }
            node = node.descendant;
        }
        return false;
    }

}


class Node<T> {

    T value;
    Node descendant;

    public Node(T value){

        this.value = value;
    }
}

File: DataStructures/LinkedList/LinkedList.py
#!/usr/bin/env python3


class Node:

    def __init__(self, value):
        self.value = value
        self.descendant = None


class LinkedList(object):

    def __init__(self, value=None):

        if value is None:
            self.root = None
            return
        if type(value).__name__ == 'list':
            if len(value) == 0:
                self.root = None
            else:
                self.root = Node(value[0])
                for i in value[1:]:
                    self.Add(i)


        else:
            self.root = Node(value)

    def AddToBeginning(self, value):
        node = Node(value)
        node.descendant = self.root
        self.root = node

    def Add(self, value):
        node = self.root
        while node.descendant is not None:
            node = node.descendant
        node.descendant = Node(value)

    def PrintList(self):
        print(self.createList())

    def createList(self):
        myList = []
        myNode = self.root

        while myNode is not None:
            myList.append(myNode.value)
            myNode = myNode.descendant
        return myList

    def Search(self, value):
        myNode = self.root

        while myNode is not None:
            if myNode.value == value:
                return True
            myNode = myNode.descendant
        return False

    def RemoveValue(self, value):
        node = self.root
        if node.value == value:
            self.root = self.root.descendant
            return True
        previousNode = node
        node = node.descendant
        while node is not None:
            if node.value == value:
                previousNode.descendant = node.descendant
                return True
            previousNode = node
            node = node.descendant
        return False

    def RemoveAllPositionsOfValue(self, value):
        a = True
        while a:
            a = self.RemoveValue(value)
